desktop/main.dev.ts

```diff
@@ -242,19 +244,33 @@ const createWindow = async () => {
     shell.openExternal(url);
   });
 
-  const res = await fetch(DISCOVERY_URL);
-  const initialConfig = (await res.json())[0];
   const savedNetId = StoreService.get('netSettings.netId');
-  const cleanStart = savedNetId !== initialConfig.netID;
-  const configFilePath = path.resolve(app.getAppPath(), process.env.NODE_ENV === 'development' ? './' : '../../config', 'config.json');
+  const configFilePath = path.resolve(app.getPath('userData'), 'node-config.json');
+  let netId;
+  let initialConfig;
+  let netConfig;
+  let isDevNet = false;
+  if (process.env.NODE_ENV === 'development' && process.env.DEV_NET_URL) {
+    const devConfig = await fetch(process.env.DEV_NET_URL);
+    netConfig = await devConfig.json();
+    netId = netConfig.p2p['network-id'];
+    isDevNet = true;
```

Comment:
BTW, you can assign a value to `initialConfig` here:
```
initialConfig = {
  netId,
  netName: 'Dev Net',
  explorer: '',
  dash: ''
};
```
and then get rid of ternary operators below.
Moreover, in this way we get rid of `undefined`.